src/element-style-observer.js

```diff
@@ -171,7 +168,7 @@ export default class ElementStyleObserver {
 	 * @param {string | string[]} properties
 	 * @return {void}
 	 */
-	observe (properties) {
+	async observe (properties) {
```

Comment:
@LeaVerou – A notable change is that I had to make the `observe` method async (because of the `await hasUnregisteredTransitionBug()` and `await hasTransitionRunLoopBug()` checks it contains). I _think_ that's OK because it's not returning anything, but let me know if you have any concerns about that.

Reply:
Unfortunately, making `observe` is a breaking change (see the tests; most of them have begun to fail). Have you tried to run the tests? Running any local HTTP server inside the repo folder and navigating to the `tests` page should be sufficient.

Reply:
Oh, I haven't! I will do that, thank you.

Reply:
@DmitrySharabin – I actually can't get the tests to pass, even in `main`. :/

![CleanShot 2025-05-18 at 11 49 19@2x](https://github.com/user-attachments/assets/0fad3f9e-3f47-41f6-a431-0adee662360c)

Reply:
Oh wow, they pass if I go to `localhost:8000/tests` instead of `127.0.0.1:8000/tests`. Sorry!

Reply:
Yeah, I expected that we would face these issues if we made `observe` async. I believe that after changing the API, we should update the tests to reflect the change. For now, they work with the code as if it's synchronous.

Reply:
Hmm, I'm confused @DmitrySharabin. Are you saying that the tests failing on `127.0.0.1` but not on `localhost` is somehow related to my changes? Because I'm seeing the same thing in `main`.

Also, I'm doubly confused because on `localhost` I can get all tests to pass except one (in "The display property") even after my changes. Which tests fail for you? 🤔

Reply:
> Oh wow, they pass if I go to `localhost:8000/tests` instead of `127.0.0.1:8000/tests`. Sorry!

Hm, that's weird. That shouldn't happen. At least I don't face it, and can safely use both interchangeably. 🤔

Reply:
@DmitrySharabin – I just pushed [something](https://github.com/LeaVerou/style-observer/pull/111/commits/25d479c0ac47440c297aded338637a47d64062aa) that makes `observe` synchronous again. Does this fix the tests for you? If not, which ones are failing? They're all passing here.

Reply:
> Hmm, I'm confused @DmitrySharabin. Are you saying that the tests failing on `127.0.0.1` but not on `localhost` is somehow related to my changes? Because I'm seeing the same thing in `main`.

No-no, I was talking about the PR's branch. Sorry for the confusion.

Reply:
> @DmitrySharabin – I just pushed [something](https://github.com/LeaVerou/style-observer/pull/111/commits/25d479c0ac47440c297aded338637a47d64062aa) that makes `observe` synchronous again. Does this fix the tests for you? If not, which ones are failing? They're all passing here.

Now, everything is OK. 👍